codebundles/k8s-fluxcd-helm-health/runbook.robot

```diff
@@ -0,0 +1,125 @@
+Metadata            Author    Shea Stewart
+Metadata            Display Name    Kubernetes FluxCD HelmRelease TaskSet
+Metadata            Supports    Kubernetes,AKS,EKS,GKE,OpenShift,FluxCD
+Library             RW.Core
+Library             RW.CLI
+Library             RW.Utils
+Library             RW.platform
+
+Suite Setup         Suite Initialization
+
+
+*** Tasks ***
+List all available FluxCD Helmreleases    
+    [Documentation]    List all FluxCD helmreleases that are visible to the kubeconfig.    
+    [Tags]        FluxCD     Helmrelease     Available    List
+    ${stdout}=    RW.CLI.Run Cli
+    ...    cmd=${KUBERNETES_DISTRIBUTION_BINARY} get ${RESOURCE_NAME} ${NAMESPACE} --context ${context}
+    ...    target_service=${kubectl}
+    ...    env=${env}
+    ...    secret_file__kubeconfig=${KUBECONFIG}
+    ...    render_in_commandlist=true
+    ${history}=    RW.CLI.Pop Shell History
+    RW.Core.Add Pre To Report    Helmreleases available: \n ${stdout}
+    RW.Core.Add Pre To Report    Commands Used:\n${history}
+
+Fetch All FluxCD Helmrelease Versions  
+    [Documentation]    List helmreleases and  the last attempted software version and the current running version.  
+    [Tags]        FluxCD     Helmrelease    Versions
+    ${stdout}=    RW.CLI.Run Cli
+    ...    cmd=${KUBERNETES_DISTRIBUTION_BINARY} get ${RESOURCE_NAME} ${NAMESPACE} -o=jsonpath="{range .items[*]}{'\\nName: '}{@.metadata.name}{'\\nlastAppliedRevision:'}{@.status.lastAppliedRevision}{'\\nlastAttemptedRevision:'}{@.status.lastAttemptedRevision}{'\\n---'}{end}" --context ${context} || true
+    ...    target_service=${kubectl}
+    ...    env=${env}
+    ...    secret_file__kubeconfig=${KUBECONFIG}
+    ...    render_in_commandlist=true
+    ${history}=    RW.CLI.Pop Shell History
+    RW.Core.Add Pre To Report    Helmreleases status errors: \n ${stdout}
+    RW.Core.Add Pre To Report    Commands Used:\n${history}
+
+Fetch Mismatched FluxCD HelmRelease Version
+    [Documentation]    List helmreleases and use jq to display any releases where the last attempted software revision doesn't match the current running revision. Requires jq.  
+    [Tags]        FluxCD     Helmrelease    Version    Mismatched    Unhealthy
+    ${stdout}=    RW.CLI.Run Cli
+    ...    cmd=${KUBERNETES_DISTRIBUTION_BINARY} get ${RESOURCE_NAME} ${NAMESPACE} -o json --context ${context} | jq -r '.items[] | select(.status.lastAppliedRevision!=.status.lastAttemptedRevision) | "Name: " + .metadata.name + " Last Attempted Version: " + .status.lastAttemptedRevision + " Last Applied Revision: " + .status.lastAppliedRevision'
+    ...    target_service=${kubectl}
+    ...    env=${env}
+    ...    secret_file__kubeconfig=${KUBECONFIG}
+    ...    render_in_commandlist=true
+    RW.CLI.Parse Cli Output By Line
+    ...    rsp=${stdout}
+    ...    set_issue_title="FluxCD Helmrelease Version Mismatch"
+    ...    lines_like_regexp=".+"
```

Comment:
example of parsing by line: https://github.com/runwhen-contrib/rw-cli-codecollection/blob/main/codebundles/cli-test/runbook.robot#L48

Some notes regarding regex when dealing with robot + python:
- Robot has a nasty habit of dropping quotes in strings, even with raw strings. I've yet to find a good way to tackle this without manually adding extra quotes which obviously removes the 1:1 nature of taking your tested regex and pasting it in...
- Python's regex standard differs ever so slightly. An example of this is in the example link, the generic regex language doesnt use the `P` character to function, but python needs it.
- I use https://regexr.com/ for testing them, and then adjusting for python where needed + add extra quotes to deal with robot

Reply:
Besides your extra double quotes, it looks like you've got it right btw. Issues will have default titles, descriptions, etc if you dont set them as I generate them based on the comparisons you provide so they're very generic / mathematical. As you've done here, you can personalize them by explicitly setting them in the parse call :)

Reply:
I think my last commit should resolve this in a way that mostly makes sense to me (and works as tested against another codebundle I'm working on)